src/jit/morph.cpp

```diff
@@ -3915,37 +3895,20 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
                     // We only need a copy if we have one of the following:
                     // - We have a lclVar that has been promoted and is passed in registers.
                     // - The sizes don't match.
-                    // - We have a vector intrinsic.
-                    // TODO-Amd64-Unix-CQ: The first and last case could and should be handled without copies.
+                    // TODO-Amd64-Unix-CQ: The first case could and should be handled without copies.
 
                     copyBlkClass = NO_CLASS_HANDLE;
 
-                    // TODO-Amd64-Unix-CQ: This should use the condition below, which captures whether it is actually
-                    // being passed in registers (not just that the struct is eligible if there are enough regs left).
-                    // Also, this way we don't need to keep the structDesc in the argEntry if it's not actually passed
-                    // in registers.
                     // TODO-Amd64-Unix-Throughput: We don't need to keep the structDesc in the argEntry if it's not
                     // actually passed in registers.
                     if (argEntry->isPassedInRegisters())
                     {
                         assert(argEntry->structDesc.passedInRegisters);
-                        if ((lclVar != nullptr) &&
-                            (lvaGetPromotionType(lclVar->gtLclVarCommon.gtLclNum) == PROMOTION_TYPE_INDEPENDENT))
-                        {
-                            copyBlkClass = objClass;
-                        }
-                        else if (passingSize != structSize)
+                        if (((lclVar != nullptr) && (lvaGetPromotionType(lclVar->gtLclVarCommon.gtLclNum) == PROMOTION_TYPE_INDEPENDENT)) ||
+                            ((argObj->OperIs(GT_OBJ)) && (passingSize != structSize)))
```

Comment:
Remove parenthesis around `argObj->OperIs(GT_OBJ)`. This is IMO one of advantages of using `OperIs` over using `OperGet() == GT_OBJ`.